internal/storage/sql/common/rollout.go

```diff
@@ -323,64 +323,78 @@ func (s *Store) CreateRollout(ctx context.Context, r *flipt.CreateRolloutRequest
 			Id:           uuid.Must(uuid.NewV4()).String(),
 			NamespaceKey: r.NamespaceKey,
 			FlagKey:      r.FlagKey,
-			Type:         r.Type,
 			Rank:         r.Rank,
 			Description:  r.Description,
 			CreatedAt:    now,
 			UpdatedAt:    now,
 		}
 	)
 
+	switch r.GetRule().(type) {
+	case *flipt.CreateRolloutRequest_Segment:
+		rollout.Type = flipt.RolloutType_SEGMENT_ROLLOUT_TYPE
+	case *flipt.CreateRolloutRequest_Percentage:
+		rollout.Type = flipt.RolloutType_PERCENTAGE_ROLLOUT_TYPE
+	default:
+		return nil, fmt.Errorf("invalid rollout rule type %T", r.GetRule())
+	}
+
```

Comment:
It seems like we are doing the `switch` statement twice in this `CreateRollout` method?

Reply:
We have to because we need to derive the type in order to insert it.
We need to insert the flag before the rollouts, because the rules need the rollout ID to exist on the rollout table for the integrity contraints to pass.

Reply:
I'm good w keeping the switch